Update FH_auto_classes.py

gta4hater opened the pull request:
I noticed an bug where it shifted into reverse when driving off (from 0km/h) at only 6% throttle(also underhard breaking sometimes) and noticed that it was because i didnt change the variable so i did that instead and reverted back to the normal downshifting pattern again. The bug fix for lowerrpm cars is still in place!

gta4hater committed:
Update FH_auto_classes.py

I noticed an bug where it shifted into reverse when driving at only 6% throttle and noticed that it was because i didnt change the variable so i did that instead and reverted back to the normal downshifting pattern again. The bug fix for lowerrpm cars is still in place!

gta4hater committed:
Update FH_auto_classes.py

made other changes for ford crown vic which has max_rpm_top of 6149 rpm lol

GinoLin980 commented on src/FH_auto_classes.py:
```diff
@@ -136,7 +136,11 @@ def analyzeInput(self) -> None:
+        self.rpm_range_bottom = (
+        max(idleRPM + (min(self.gear, 6) * 70) * 0.96, self.rpm_range_top - self.rpm_range_size) - 400 
+        if (self.rpm_range_top <= 6100) 
+        else max(idleRPM + (min(self.gear, 6) * 70) * 0.96, self.rpm_range_top - self.rpm_range_size)
```
can be simplified for better readability

```python
        self.rpm_range_bottom = max(idleRPM + (min(self.gear, 6) * 70) * 0.96, self.rpm_range_top - self.rpm_range_size)
        if (self.rpm_range_top <= 6100):
            self.rpm_range_bottom -= 400
```

GinoLin980 commented:
> I noticed an bug where it shifted into reverse when driving off (from 0km/h) at only 6% throttle(also underhard breaking sometimes) and noticed that it was because i didnt change the variable so i did that instead and reverted back to the normal downshifting pattern again. The bug fix for lowerrpm cars is still in place!

@gta4hater 
Do you think this could be triggered by `kickdown` feature? I met that problem before but it was when developing `kickdown` feature. But I can't see anything causing it from the current code.

Will ignoring user input(gas, brake) for a few seconds(or an amount of data) be a good idea to prevent this situation.

And I think there's **latency between local network communication and processing latency**, which might result in this problem at hard braking.(maybe we're already from 2nd gear to 1st gear in game. but the program still receives 2nd gear due to the latency, therefore it performs down shift again?)

GinoLin980 commented:
Oh, you mean you have fixed this situation?

gta4hater commented:
> > I noticed an bug where it shifted into reverse when driving off (from 0km/h) at only 6% throttle(also underhard breaking sometimes) and noticed that it was because i didnt change the variable so i did that instead and reverted back to the normal downshifting pattern again. The bug fix for lowerrpm cars is still in place!
> 
> @gta4hater Do you think this could be triggered by `kickdown` feature? I met that problem before but it was when developing `kickdown` feature. But I can't see anything causing it from the current code.
> 
> Will ignoring user input(gas, brake) for a few seconds(or an amount of data) be a good idea to prevent this situation.
> 
> And I think there's **latency between local network communication and processing latency**, which might result in this problem at hard braking.(maybe we're already from 2nd gear to 1st gear in game. but the program still receives 2nd gear due to the latency, therefore it performs down shift again?)

i printed the gears and the latency was spot, it was instant as far i know (i printed the gear and throttle). The issue was in the downshift logic from my if statement. I changed it because after the change the issue where it shfited into reverse didnt happend. 

which makes sense because the other function didnt have the new variable for the lower rpm cars so the kickdown "function" had the old variable and made it shift one more gear down? well idk now it works lol

gta4hater commented:
> Oh, you mean you have fixed this situation? Nevermind, I'll leave this thought here if anyone needs it in the future :)

yep

GinoLin980 commented:
> > Oh, you mean you have fixed this situation? Nevermind, I'll leave this thought here if anyone needs it in the future :)
> 
> yep 

I made a comment on your code. 
Please have a look. 
https://github.com/GinoLin980/Forza-Horizon-realistic-gearbox/pull/5#discussion_r1819938385

gta4hater committed:
Update FH_auto_classes.py

made it simpler

gta4hater commented:
im not the best in python ngl :(((

GinoLin980 commented:
It's fine my brother. 
Imma merge this PR.
